__tests__/helpers.js

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+const MemoryFileSystem = require("memory-fs");
+const webpack = require("webpack");
+
+exports.PluginEnvironment = class PluginEnvironment {
```

Comment:
from [here](https://github.com/webpack/webpack/blob/master/test/helpers/PluginEnvironment.js)

The `memory-fs` is being used to run a webpack compile without needing to emit files for the test.